docs/content/sips/00x-cloudevent-trigger.md

```diff
@@ -0,0 +1,263 @@
+2. Establish reasonable performance metrics for the new CloudEvents component.
+
+## Future design considerations
+
+#### More transport protocols bindings
+- Kafka binding
+- AMQP binding
+- NATS binding
+
+#### Filtering based on event attributes
+```toml
+[[component]]
+id = "filter"
+source = "target/wasm32-wasi/release/spinhelloworld.wasm"
+description = "A simple component that filters events based on event attributes."
+schema = "cloudevents"
+[component.trigger]
+route = "/filter"
+[component.filter]
+ce.type = "com.example.someevent"
+ce.source = "/mycontext/subcontext"
+```
+
+#### Generic CloudEvent component
+A generic CloudEvents component is defined in the following way:
+```rust
+// A Spin CloudEvents component written in Rust
+use anyhow::Result;
+use spin_sdk::{
+    event::{Event},
+    cloud_event_component,
+};
+
+/// A simple Spin event component.
+#[cloud_event_component]
+fn trigger(event: Event) -> Result<Event, _> {
+    println!("event is {}", event.id());
+    // do something with the event
+    Ok(event)
+}
+```
+
+It is trigger-agnostic, at least within the supported CloudEvents protocol bindings. You can see the list of supported protocols in the [CloudEvents documentation](https://github.com/cloudevents/spec/blob/main/cloudevents/bindings). The benefits of doing this are:
+1. Rapid prototyping: you can quickly prototype your event components and test them locally using HTTP bindings. Once you are confident that they are working, You can switch the trigger to a different type, without having to modify the code.
+2. Reusability: you can reuse the same event components with different protocols bindings, such as AMQP and Kafka.
+3. Chaining: you can chain multiple event components together since they share the same component signature.
+
+#### CloudEvents trigger
+
+Creating an CloudEvents trigger is done when [configuring the application](/configuration)
+by defining the top-level application trigger:
+
+```toml
+# spin.toml
+trigger = { type = "cloudevent" }
+```
+
+Then, when defining the component (in `spin.toml`), you can set the protocol binding for the component. For example:
+
+- an HTTP CloudEvents component:
+
+```toml
+[component.trigger]
+binding = "http"
+```
+
+- an Kafka CloudEvents component (optional):
+
+```toml
+[component.trigger]
+binding = "kafka"
+broker = ["localhost:9092", "localhost:9093"]
+topic = "mytopic"
+group = "mygroup"
+```
+
+- an AMQP CloudEvents component (optional WIP):
+
+```toml
+[component.trigger]
+binding = "amqp"
+broker = "localhost:5672"
+exchange = "myexchange"
+routing_key = "myroutingkey"
+```
+
+You can also set the sink address for the component that returns a CloudEvent. For example:
+
+```toml
+[component.trigger]
+binding = "http"
+sink = "http://localhost:8080/someresource"
+```
+
+Note that the sink address is only used when the component is invoked. The component will make a outbound HTTP request that includes the CloudEvents to the sink address.
```

Comment:
Is a "sink address" specific for HTTP?
What is the default way of returning a result from a CloudEvents invocation?

For example, if there is an invocation triggered by an AMQP event, what is the expectation for the result?
Should the component itself make outbound requests (be those HTTP, or publish messages), or is it expected that the underlying runtime should do something with the returned event data?